lib/rules/no-else-return.js

```diff
@@ -88,6 +102,34 @@ module.exports = {
                     }
 
                     const endToken = sourceCode.getLastToken(node);
+
+                    /**
+                     * If else block includes block scope local variable declaration [let or const],
+                     * then it is not safe to remove else keyword [issue 11069]
+                     */
+                    const isDeclarationInside = sourceCode.getTokensBetween(startToken, endToken).findIndex(isVariableDeclaration) > -1;
+
+                    const childScopeVars = (childScopes, store = {}, level = 0) =>
+                        childScopes.reduce((acc, block) => {
+                            if (block.childScopes.length >= 0) {
+                                acc[level] = block.variables;
+                                let index = level + 1
+                                return childScopeVars(block.childScopes, acc, index);
+                            }
+                            return acc
+                        }, store)
+
+                    const shadowedVariables = scope => {
+                        const result = childScopeVars(scope.childScopes);
+                        return scope.variables.filter(elem => result[0].findIndex(vars => vars.name === elem.name) > -1)
+                    };
+                    const shadowed = shadowedVariables(context.getScope());
+                    console.log('test', shadowed)
+
+                    if (isDeclarationInside) {
```

Comment:
this will evaluate to `true` if there's a `let` or `const` somewhere in a nested child scope. instead it should only abort here if the `else`-block contains block scoped declarations

Reply:
Does it mean, that if `else` contain block scoped declarations , `fix` method has to return null (abort)? 
Can you please provide an example?

Reply:
"Block scoped declaration" means `const`, `let` and `class` as they are not hoisted to the containing function.

```ts
let foo = 1;
function canNotBeFixed() {
  if (condition) {
    return foo;
  } else {
    let foo = 2; // else-block contains block scoped declaration, can not be fixed
    console.log(foo);
  }
}

function canBeFixed() {
  if (condition) {
    return foo;
  } else {
    {
      let foo = 2; // block scoped declaration is not directly in else-block, but in a nested block
      console.log(foo);
    }
  }
}
```